pandas/core/categorical.py

```diff
@@ -443,12 +443,18 @@ def _validate_categories(cls, categories):
             raise ValueError('Categorical categories must be unique')
         return categories
 
-    def _set_categories(self, categories):
+    def _set_categories(self, categories, validate=True):
         """ Sets new categories """
```

Comment:
that's my point - if I am fast pathing I don't need any validation at all (so I want to call _set_categories) to assign them but not validate

rather than

self._categories = ...